fv3core/stencils/fxadv.py

```diff
@@ -1,314 +1,381 @@
+from gt4py.gtscript import PARALLEL, computation, horizontal, interval, region
 
-import fv3core._config as spec
-import fv3core.utils.gt4py_utils as utils
 from fv3core.decorators import gtstencil
+from fv3core.utils.typing import FloatField
 
-from ..utils import global_config
 
+@gtstencil()
+def fxadv_stencil(
+    cosa_u: FloatField,
+    cosa_v: FloatField,
+    rsin_u: FloatField,
+    rsin_v: FloatField,
+    sin_sg1: FloatField,
+    sin_sg2: FloatField,
+    sin_sg3: FloatField,
+    sin_sg4: FloatField,
+    rdxa: FloatField,
+    rdya: FloatField,
+    area: FloatField,
+    dy: FloatField,
+    dx: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+    crx_adv: FloatField,
+    cry_adv: FloatField,
+    xfx_adv: FloatField,
+    yfx_adv: FloatField,
+    ut: FloatField,
+    vt: FloatField,
+    ra_x: FloatField,
+    ra_y: FloatField,
+    dt: float,
+):
+    """
+    Updates fluxes and
+    Inputs:
+        uc: x-velocity on the C-grid
+        vc: y-velocity on the C-grid
+        crx_adv: Courant number, x direction(inout)
+        cry_adv: Courant number, y direction(inout)
+        xfx_adv: Finite volume flux form operator in x direction (inout)
+        yfx_adv: Finite volume flux form operator in y direction (inout)
+        ut: temporary x-velocity transformed from C-grid to D-grid equivalent(?) (inout)
+        vt: temporary y-velocity transformed from C-grid to D-grid equivalent(?) (inout)
+        ra_x: Area increased in the x direction due to flux divergence (inout)
+        ra_y: Area increased in the y direction due to flux divergence (inout)
+        dt: timestep in seconds
+    Grid variables inputs:
+        cosa_u, cosa_v, rsin_u, rsin_v, sin_sg1,sin_sg2, sin_sg3, sin_sg4:
+        rdxa, rdya, area, dy, dx
+    Returns:
+       Updates to xfx_adv, yfx_adv, crx_adv, cry_adv, ut, vt, ra_x, ra_y
+    """
+    from __externals__ import local_ie, local_is, local_je, local_js
 
-def grid():
-    return spec.grid
-
+    with computation(PARALLEL), interval(...):
+        ut = main_ut(uc, vc, cosa_u, rsin_u, ut)
+        ut = ut_y_edge(uc, sin_sg1, sin_sg3, ut, dt)
+        vt = main_vt(uc, vc, cosa_v, rsin_v, vt)
+        vt = vt_y_edge(vc, cosa_v, ut, vt)
+        vt = vt_x_edge(vc, sin_sg2, sin_sg4, vt, dt)
+        ut = ut_x_edge(uc, cosa_u, vt, ut)
+        ut = ut_corners(uc, vc, cosa_u, cosa_v, ut, vt)
+        vt = vt_corners(uc, vc, cosa_u, cosa_v, ut, vt)
+    with computation(PARALLEL), interval(...):
+        with horizontal(region[local_is : local_ie + 2, :]):
+            prod = dt * ut
+            crx_adv = prod * rdxa[-1, 0, 0] if prod > 0 else prod * rdxa
+            xfx_adv = dy * prod * sin_sg3[-1, 0, 0] if prod > 0 else dy * prod * sin_sg1
+    with computation(PARALLEL), interval(...):
+        with horizontal(region[local_is : local_ie + 2, :]):
+            ra_x = ra_x_func(area, xfx_adv)
+    with computation(PARALLEL), interval(...):
```

Comment:
Seeing the multiple computation and intervals at first made me think there is vertical dependence in this stencil, but I can't find any. Could you put this all into the same computation interval? Shouldn't matter from a correctness perspective, since it's merged anyway in the compilation process.

Reply:
Good point, these different computations were previously in different stencils, and do not actually need to be different computations

Reply:
Oh actually sorry, just remembered -- we do need a computation separating the ra variables from the xfx, as the ra uses xfx at horiontal offsets. But I can reduce this down to 2 computations rather than 4